llvm/unittests/Object/SYCLBINTest.cpp

```diff
@@ -0,0 +1,259 @@
+#include "llvm/Object/SYCLBIN.h"
+#include "llvm/SYCLPostLink/ModuleSplitter.h"
+#include "llvm/Support/FileOutputBuffer.h"
+
+#include "llvm/Testing/Support/Error.h"
+#include "gtest/gtest.h"
+#include <random>
+
+using namespace llvm;
+using namespace llvm::sys;
+using namespace llvm::object;
+using namespace llvm::util;
+
+#define ASSERT_NO_ERROR(x)                                                     \
+  if (std::error_code ASSERT_NO_ERROR_ec = x) {                                \
+    SmallString<128> MessageStorage;                                           \
+    raw_svector_ostream Message(MessageStorage);                               \
+    Message << #x ": did not return errc::success.\n"                          \
+            << "error number: " << ASSERT_NO_ERROR_ec.value() << "\n"          \
+            << "error message: " << ASSERT_NO_ERROR_ec.message() << "\n";      \
+    GTEST_FATAL_FAILURE_(MessageStorage.c_str());                              \
+  } else {                                                                     \
+  }
+
+namespace {
+// Helper class for ensuring the destruction of
+class ManagedBinaryFile {
+public:
+  ManagedBinaryFile(ManagedBinaryFile &&) = default;
+  ManagedBinaryFile(const ManagedBinaryFile &) = delete;
+
+  ~ManagedBinaryFile() { fs::remove(FileName); }
+
+  static Expected<ManagedBinaryFile> create(StringRef FileName,
+                                            const std::vector<uint8_t> &Data) {
+    Expected<std::unique_ptr<FileOutputBuffer>> OutputOrErr =
+        FileOutputBuffer::create(FileName, Data.size());
+    if (!OutputOrErr)
+      return OutputOrErr.takeError();
+    ManagedBinaryFile ManagedFile{FileName};
+    std::unique_ptr<FileOutputBuffer> Output = std::move(*OutputOrErr);
+    llvm::copy(Data, Output->getBufferStart());
+    if (Error E = Output->commit())
+      return std::move(E);
+
+    return std::move(ManagedFile);
+  }
+
+private:
+  ManagedBinaryFile(StringRef FileName) : FileName{FileName.str()} {}
+
+  std::string FileName;
+};
+
+std::vector<uint8_t> generateRandomImage() {
+  std::mt19937 Rng(std::random_device{}());
```

Comment:
Do you think it's possible for any of the randomness used here to cause a test to fail? If not, that's fine, but if so, tests with random failures are an absolute nightmare to debug

Reply:
I really doubt it. This was just adopted from other surrounding object test cases. I don't mind testing with some static image data if that is preferred, but generally speaking the contents of the image should not matter to SYCLBIN.